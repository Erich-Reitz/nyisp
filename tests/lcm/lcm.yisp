(define mod (a b)
    (cond ((< a b) a)
          (t (mod (- a b) b))))

(define gcd (a b)
    (cond ((= b 0) a)
          (t (gcd b (mod a b)))))

(define lcm (a b)
  (/ (* a b) (gcd a b)))


(define lcm-list (lst)
  (cond 
    ((NIL? (cdr lst)) (car lst))
    (t (lcm (car lst) (lcm-list (cdr lst)))))
)

(lcm-list '(2 3))
(lcm-list '(2 32))
(lcm-list '(9 7))
(lcm-list '(4 5))