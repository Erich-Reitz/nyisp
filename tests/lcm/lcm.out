gcd
lcm
lcm-list
6
32
63
20
